src/Signer/Eddsa.php

```diff
@@ -0,0 +1,49 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Signer;
+
+use Lcobucci\JWT\Signer;
+use SodiumException;
+
+use function extension_loaded;
+use function sodium_crypto_sign_detached;
+use function sodium_crypto_sign_verify_detached;
+
+final class Eddsa implements Signer
+{
+    /** @throws SodiumExtensionMissing */
+    public function __construct()
+    {
+        if (! extension_loaded('sodium')) {
+            // @codeCoverageIgnoreStart
+            throw new SodiumExtensionMissing(
+                'The extension "sodium" is not available. Please install it to use this method.'
+            );
+            // @codeCoverageIgnoreEnd
+        }
+    }
+
+    public function algorithmId(): string
+    {
+        return 'EdDSA';
+    }
+
+    public function sign(string $payload, Key $key): string
+    {
+        try {
+            return sodium_crypto_sign_detached($payload, $key->contents());
+        } catch (SodiumException $sodiumException) {
+            throw new InvalidKeyProvided($sodiumException->getMessage(), 0, $sodiumException);
```

Comment:
I've used `InvalidKeyProvided` exception but not its factory methods because `SodiumException` has much wider scope range

Reply:
We could introduce a named constructor there but this works too.